pages/index.tsx

```diff
@@ -36,42 +38,58 @@ export default function Home({ deviceType }: Props) {
   const coordinatesAndEventType:
     | CoordinatesURLParametersWithEventType
     | undefined = useCoordinates();
+  const urlParams = useMemo(
+    () =>
+      new URLSearchParams({
+        ne_lat: coordinatesAndEventType?.ne_lat,
+        ne_lng: coordinatesAndEventType?.ne_lng,
+        sw_lat: coordinatesAndEventType?.sw_lat,
+        sw_lng: coordinatesAndEventType?.sw_lng,
+      } as any).toString(),
+    [
+      coordinatesAndEventType?.ne_lat,
+      coordinatesAndEventType?.ne_lng,
+      coordinatesAndEventType?.sw_lat,
+      coordinatesAndEventType?.sw_lng,
+    ]
+  );
+  const { error, isLoading, mutate, isValidating } = useSWR<
+    MarkerData[] | undefined
+  >(url, dataFetcher, {
+    onLoadingSlow: () => setSlowLoading(true),
+    revalidateOnFocus: false,
+  });
+  const { setDevice } = useMapActions();
+  const [remainingTime, resetThrottling] = useIncrementalThrottling(
+    mutate,
+    REQUEST_THROTTLING_INITIAL_SEC
+  );
 
-  const urlParams = new URLSearchParams({
-    ne_lat: coordinatesAndEventType?.ne_lat,
-    ne_lng: coordinatesAndEventType?.ne_lng,
-    sw_lat: coordinatesAndEventType?.sw_lat,
-    sw_lng: coordinatesAndEventType?.sw_lng,
-  } as any).toString();
-
-  function handleButtonClick() {
+  const handleScanButtonClick = useCallback(() => {
     setURL(BASE_URL + "?" + urlParams);
+    resetThrottling();
```

Comment:
Bunu soyle mi yapsak acaba cunku `setURL` sync degil. Tam inceleyemedim `resetThrottling` implementationi ama PR merge edilmeden bu yorumu atmak istedim. 

```
useEffect(() => {
   resetThrottling();
}, [url]) 
```

Reply:
Manuel olarak butona tıklandığında trigger etmek istemiyor muyuz burada?

Reply:
Veya butona tıkladıktan sonra bi süre disabled tutabiliriz

Reply:
Butona tikladigimizda URL set ediliyor zaten, eger URL Degisirse de o da bunu tetikleyecek zaten. 

Biraz daha detayli bakayim en iyisi. Kafam karisti biraz.

Reply:
https://deprem-yardim-frontend-git-fork-keislamoglu-feat-453-afetharita.vercel.app/

Buradan check edebilirsin davranışı

Reply:
Butona tiklandiginda cache'i invalidate etmeyebiliriz belki de. Sonucta bu veri her saniye degisen bir sey degil. 

Yani `mutate()` i kaldirabiliriz? Su anda bu sebeple 2 request yapiyoruz

1) `mutate` sonucu (URL o anda guncellenmediginde onceki URLye bir daha atiyoruz) 
2) `setURL` sonucu key degistigi icin. 

1.yi yapmiyor olmamiz gerekiyor.

Reply:
Eger dilerseniz beraber review edebiliriz sesli kanalda.

Reply:
Merge sonrası akışı iyi takip edememişim, şimdi oturdu bende. Dikkatin için teşekkürler 🙏